src/index.ts

```diff
@@ -162,7 +164,7 @@ function getCaseClassDeclaration(caseClass: CaseClass): Reader<Ctx, gen.TypeDecl
   return traverseReader(caseClass.members, getProperty).map(properties => {
     const interfaceDecl = gen.typeCombinator(properties, caseClass.name);
     if (caseClass.typeParams && caseClass.typeParams.length > 0) {
-      const staticParams = caseClass.typeParams.map(p => `${p.name} extends t.Any`).join(', ');
+      const staticParams = caseClass.typeParams.map(p => `${p.name} extends t.Mixed`).join(', ');
```

Comment:
not strictly related but while at it... `t.Mixed` should be in general safer than `t.Any`, although I don't think it makes any difference in this case

Reply:
isn't there a `t.unknown` we can use?